### ISSUE TITLE
Builds/packages for Linux

### ISSUE THREAD

**Issue by redmanmale** [uTox/uTox#701]
I'm trying to move couple of friend to uTox and they use Ubuntu.
They are not programmers, so it would really hard (if not impossible) for them to build uTox from source.

But I could not find any up-to-date binaries of uTox!

In README and Tox.io site there are links to that [artifact (build.tox.chat)](https://build.tox.chat/view/uTox/job/uTox_build_linux_x86-64_release/lastSuccessfulBuild/artifact/utox_linux_x86-64.tar.xz), but inside is 0.9.7.
In last several GitHub releases there are different kinds of builds for Windows, but none for Linux.
Also there are not any mentions for ppa or some other kind of packages distribution.

I know Linux people often compiling from sources themselves, but still...

**Comment by GrayHatter** [comment]
Having battled with this for the past 2hours. No, I'm done.

We're just going to refer to system packagers. 
https://www.archlinux.org/packages/community/x86_64/utox/
https://tox.chat/download.html#gnulinux
https://www.freshports.org/net-im/uTox

@robinlinden was looking at OBS https://build.opensuse.org/ maybe he could be convinced to set something up

**Comment by robinlinden** [comment]
There are a lot of Ubuntu builds on [build.tox.chat](https://build.tox.chat/view/uTox/)
You might want to sort by last successful build to find them.

**Comment by GrayHatter** [comment]
@Encrypt How much disk space, and ram usage does your build script take up? Perhaps we can migrate that to https://utox.io as well?

**Comment by GrayHatter** [comment]
It's already running... *shudders* ...debian, so it should be easy-ish to set up and get working.

**Comment by redmanmale** [comment]
Finally I got latest package version from [here](https://tox.chat/download.html#gnulinux).

Latest successful build for linux: https://build.tox.chat/view/uTox/job/uTox_build_linux_x86-64_release/lastSuccessfulBuild was Build 143 (Sep 28, 2016 7:39:47 PM).
Also I found https://build.tox.chat/view/uTox/job/uTox_pkg_linux_deb_shared_xenial_x86-64_stable_release/ and it's pretty up-to-date, but until now I had no idea to search for this name.

IMO, plain "linux_x86-64" is the "main" artefact, which I was looking for.

**Comment by GrayHatter** [comment]
@redmanmale right, a few months ago, everyone involved with pkg.tox.chat agreed that we should be shipping shared binaries instead of static binaries when we can. At that point I unofficially stopped supporting the static linux builds on tox.chat, and turned over the shared bins to @Encrypt who's been doing a lot of good work across ALL the clients.

**Comment by redmanmale** [comment]
@GrayHatter 
I got it, sorry for making such a fuss here.

**Comment by GrayHatter** [comment]
@redmanmale no reason to be sorry. We should be making it easy to get uTox, and clearly it's not easy yet.

**Comment by Encrypt** [comment]
Hello @redmanmale!

The best way to install uTox on Ubuntu is definitely to use the official repository.
As you pointed out, the instructions are given here: https://tox.chat/download.html#gnulinux

As presented under the "Debian/Ubuntu" section, there are two kind of releases:
- Stable releases, following the [master/](https://github.com/uTox/uTox/tree/master) branch.
- Nightly releases, following the [develop/](https://github.com/uTox/uTox/tree/develop) branch.

Stable builds are produced when there are "sufficient" changes in the develop/ branch and when the code is stable.
Nightly builds are produced whenever a pull request is merged, therefore if you want a bleeding edge client with the latest features (but also potentially more bugs :stuck_out_tongue:), you could use these packages.

Cheers!

**Comment by GrayHatter** [comment]
>  I don't think it would be relevant to migrate that to utox.io... It is better to keep everything on Jenkins, I think.

I'm trying to get away from depending on build.tox.chat so It'd be better to move it to utox.io, unless you have a better reason it should stay on build.tox.chat?

@robinlinden did you ever look into creating a OBS setup?

**Comment by robinlinden** [comment]
I had a brief look, but I don't think I have enough time to maintain all of that when I don't know much about it. Maybe we can ask Anton Batenev to take it up again?

**Comment by GrayHatter** [comment]
We have Arch, we have Gentoo, toxdeb would give us Ubuntu and Debian. We probably could use alien to convert .deb to .rpm hitting OpenSUSE, CentOS, and Fedora.

Do any others matter? Honestly, I'm not that concerned about Linux. They have already solved the issue of getting and building sources. And if the community cared much about easily getting bins, they'd have a solution like ABS for Arch, and Gentoo's inbuild with portage.

**Comment by GrayHatter** [comment]
Keeping in mind that there's more people using WinXP, than EVERY dist of *NIX

**Comment by GrayHatter** [comment]
Anyone following, we're now looking into building with muslc for distribution. 

But it's not to high on the priority list unless more people comment they want "official" static linux builds

**Comment by robinlinden** [comment]
Until we start distributing static musl libc builds of uTox the static Linux builds on build.tox.chat are working again.
[32-bit stable](https://build.tox.chat/view/uTox/job/uTox_build_linux_x86_release/)
[64-bit stable](https://build.tox.chat/view/uTox/job/uTox_build_linux_x86-64_release/)

**Comment by redmanmale** [comment]
How about store latest release (tag) version at the https://downloads.utox.io/stable?
And set this link to official site (https://utox.io/#downloads Linux).

Because now the only one link from site there leads to ArchLinux site.

**Comment by GrayHatter** [comment]
@NicoHood should really update uTox for arch linux :/

**Comment by NicoHood** [comment]
@GrayHatter Done

**Comment by redmanmale** [comment]
Maybe we should copy info about getting uTox via package systems (from https://tox.chat/download.html#gnulinux) at the uTox site right next to the Arch?

![pic](https://puu.sh/vSb8p/8ae73bbac8.png)

P.S. I don't know is it important or not, but uTox version at the [freshports.org](https://www.freshports.org/net-im/uTox) is really outdated (0.11.1).

**Comment by redmanmale** [comment]
**Bump**

Since ppa for Ubuntu/Debian has died this is important more than ever.

Right now there's no way to get a **release** (it's 0.16.1) build for Linux.
https://build.tox.chat provides only latest development build and doesn't store releases.

Why don't we provide builds for Linux (and Windows too) in [Github releases](https://github.com/uTox/uTox/releases/tag/v0.16.1)? It would be really VERY useful. One place to get all sources and binaries alike. For all systems.

@GrayHatter @robinlinden @endoffile78 
Thoughts?

**Comment by avoidr** [comment]
Generally, I don't think distributing Linux binaries is the right thing
to do, but I figure that in our situation it would be more helpful than
harmful until we are stable (So for a long time, still).

If someone wants to build a binary, it will have to be 100% statically
linked.

**Comment by lmaddox** [comment]
I'm not a Debian maintainer/packager, so I'm a bit stuck packaging these. My `utox.sh` is crashing when linking the package. 

- [this](https://github.com/Encrypt/toxdeb.git) needs to be archived, as it's completely unmaintained. The README is wrong. The built-in `help` function is wrong.
- [this](https://build.tox.chat/view/uTox/job/uTox_pkg_linux_deb_shared_xenial_x86-64_stable_release/) needs server maintenance. My browser can't connect, even after ignoring the SSL warnings.

my code (also borked):
- [c-toxcore.sh.txt](https://github.com/user-attachments/files/16402121/c-toxcore.sh.txt)
- [filter-audio.sh.txt](https://github.com/user-attachments/files/16402122/filter-audio.sh.txt)
- [utox.sh.txt](https://github.com/user-attachments/files/16402123/utox.sh.txt)
- [Dockerfile.txt](https://github.com/user-attachments/files/16402124/Dockerfile.txt)

Anyway, have fun. I'm done messing with it.
